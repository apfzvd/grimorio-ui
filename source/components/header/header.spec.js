import '../../../internals/test/helper';

import Header from './header-component';

/** @test {Header} */
describe('Header component', () => {
  const emptyProps = {
    children: undefined,
    showLogo: undefined,
    onLogoClick: undefined,
  }

  let wrapper

  beforeAll(() => {
    wrapper = shallow(
      <Header {...emptyProps} />
    );
  })

  describe('render', () => {
    it('Basic', () => {
      wrapper.setProps({ ...emptyProps, children: <div>teste com filho</div> })
      expect(wrapper.debug()).toMatchSnapshot();
    });

    it('No logo version', () => {
      wrapper.setProps({ ...emptyProps, showLogo: false })
      expect(wrapper.debug()).toMatchSnapshot();
    });

    it('Showing Logo', () => {
      wrapper.setProps({ ...emptyProps, showLogo: true })
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });

  it('onClick logo mobile, should call onLogoClick props', () => {
    const props = {
      onLogoClick: jest.fn()
    }
    const spy = jest.spyOn(props, 'onLogoClick')
    wrapper.setProps({ ...emptyProps, showLogo: true, onLogoClick: props.onLogoClick })
    wrapper.find('.logo').simulate('click')
    expect(spy).toHaveBeenCalled();
  })
})
