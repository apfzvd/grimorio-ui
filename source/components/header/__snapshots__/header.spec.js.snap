// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component render Basic 1`] = `
"<header className=\\"header\\">
  <div className=\\"content\\">
    <div>
      teste com filho
    </div>
  </div>
</header>"
`;

exports[`Header component render No logo version 1`] = `
"<header className=\\"header\\">
  <div className=\\"content\\" />
</header>"
`;

exports[`Header component render Showing Logo 1`] = `
"<header className=\\"header showLogo\\">
  <div className=\\"logo\\" onClick={[undefined]} />
  <div className=\\"content showLogo\\" />
</header>"
`;
